### PR TITLE
ISPN-5370 Make clear() non-transactional and lock free

### PR DESCRIPTION
**Pull request by pruivo**
* no concurrent operation
* no locks acquired
* no transaction

Also performed some cleanups in the touched classes

https://issues.jboss.org/browse/ISPN-5370

### CODE REVIEW

File: core/src/main/java/org/infinispan/Cache.java [hunk 1]
```diff
@@ -323,4 +322,15 @@
     */
    @Override
    CloseableIteratorSet<Entry<K, V>> entrySet();
+
+   /**
+    * Removes all mappings from the cache.
+    * <p/>
+    * <b>Note:</b> It assumes no other operation running in the cache during its invocation. If not the case, data can
```
Comment (wburns) [inline core/src/main/java/org/infinispan/Cache.java]: IMO this is pretty much the most important changes in this PR.  I wonder if we want to emphasize this more by saying something like "This should never be invoked in production unless you can guarantee no other invocations are ran concurrently."

**Comment by wburns** [comment]
Since we can't reliably

**Comment by wburns** [comment]
Sorry didn't mean to close..

**Comment by pruivo** [comment]
@wburns @galderz updated

**Comment by galderz** [comment]
Integrated, thanks @pruivo

**Pull request closed without merging**